store/tikv/lock_resolver.go

```diff
@@ -362,6 +362,11 @@ func (lr *LockResolver) resolveLocks(bo *Backoffer, callerStartTS uint64, locks
 
 			if status.primaryLock != nil && status.primaryLock.UseAsyncCommit && !exists {
 				err = lr.resolveLockAsync(bo, l, status)
+				// Treat nonAsyncCommitLock error as a soft error. It happens when a transaction
```

Comment:
Some integration test cases are needed for the conflict situation for `resolveAsync` and `fallback and reprewrite`, maybe we could do this in `TiCASE`?

Reply:
I've created PR https://github.com/pingcap/automated-tests/pull/535 to improve TiCase `TIDB_ASYNC_COMMIT_RESOLVE` to support testing this feature.